integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateTableSelection.scala

```diff
@@ -277,12 +281,21 @@ test("test PreAggregate table selection with timeseries and normal together") {
     sql(
       "create datamap agg0 on table maintabletime using 'preaggregate' as select dob,name from " +
       "maintabletime group by dob,name")
-    sql(
-      "create datamap agg1 on table maintabletime using 'preaggregate' DMPROPERTIES ('timeseries" +
-      ".eventTime'='dob', 'timeseries.hierarchy'='hour=1,day=1,month=1,year=1') as select dob," +
-      "name from maintabletime group by dob,name")
+
+  sql(
+    s"""
+       | create datamap agg1_year on table maintabletime
+       | using '$timeSeries'
+       | DMPROPERTIES (
+       | 'event_time'='dob',
+       | 'year_granularity'='1')
```

Comment:
Can u please add some test case for granularity value validation , for example when use has given value 1.5 or 2

Reply:
ok, There are some test case for value is 2, I will add some test case when use other value.